add vl53l0x time of flight sensor

VVESTM opened the pull request:
Hi, 

Here are some commits to handle the vl53l0x sensor.
This sensor is a time of flight sensor. It gives the distance between the sensor and the target.

This pull request contains : 
    * an update of sensor.h to add a distance channel sensor
    * driver source code, including an official library from ST
    * sample application

Regards
Vincent

VVESTM committed:
include: Add distance channel sensor

Add a distance channel in the sensor interface to be used with
time Of flight sensors (vl53l0X for example).This will indicate
the distance from the sensor to the target, in millimeters.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

erwango commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_i2c_win_serial_comms.c:
```diff
@@ -1,737 +0,0 @@
-/*
```
If not used, please remove from initial commit

erwango commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c:
```diff
@@ -37,7 +37,15 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "vl53l0x_i2c_platform.h"
 #include "vl53l0x_api.h"
-#include <Windows.h>
+#include <sensor.h>
```
Is that possible to add a OS abstraction source file for porting this code on Zephyr APIs?

erwango reviewed (changes requested):
Commit message of external library should include fields
Origin: <link to st website for this library>
Maintained-by: External

bogdan-davidoaia reviewed (changes requested):
Added my comments for the sensor driver and sample.

VVESTM committed:
driver: sensor: Add vl53l0x

Add driver for ST vl53l0x time of flight sensor.
This driver can be used in 2 modes :
  * proximity sensor :
      configure VL53L0X_PROXIMITY_THRESHOLD
      will return 1 if target is between sensor and threshold,
      else, 0
  * distance sensor :
      will return the distance in millimeters from sensor to target

This driver use the official STMicroelectronics library for vl53l0x.
(in ext/hal/st/lib/sensor/vl53l0x)

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM committed:
boards: disco_l475_iot1: Enable vl53l0x sensor

Add all files related to the board and sensor.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM commented:
I have pushed a new version taking into account erwango and  bogdan-davidoaia remarks.
Thanks for your help !

bogdan-davidoaia reviewed (approved):
Sensor driver and sample patches look ok to me.

erwango commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform_log.c:
```diff
@@ -18,7 +18,6 @@
  */
 
 
-#include <windows.h>
```
Remove this change, as you don't use this file

erwango commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,250 @@
+					 VL53L0X_CHECKENABLE_SIGMA_FINAL_RANGE,
+					 VL53L0X_SETUP_SIGMA_LIMIT);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckValue failed");
```
Same error message as upper.
Maybe we can use:
 "VL53L0X_SetLimitCheckValue signal rate failed" and  "VL53L0X_SetLimitCheckValue sigma failed"

erwango commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,250 @@
+				VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
+				1);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetLimitCheckEnable failed");
```
Same error message as upper.
Maybe we can use:
 "VL53L0X_SetLimitCheckEnable signal rate failed" and  "VL53L0X_SetLimitCheckEnable sigma failed"

erwango commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,250 @@
+					VL53L0X_VCSEL_PERIOD_FINAL_RANGE,
+					VL53L0X_SETUP_FINAL_RANGE_VCSEL_PERIOD);
+	if (ret) {
+		SYS_LOG_ERR("VL53L0X_SetVcselPulsePeriod failed");
```
Differentiate error  messsages

erwango reviewed (changes requested):
Minor comments.
Additionaly a README.rst would be required for sample.
Otherwise LGTM.

dbkinder reviewed (dismissed):
+1 for sample doc

VVESTM commented:
Hi,

I just rebased on the master branch because there was a merge conflict on ext/hal/Makefile.
No change at all compared to previous version.

Vincent

bogdan-davidoaia commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,250 @@
+	u8_t isApertureSpads;
+	struct vl53l0x_data *drv_data = dev->driver_data;
+
+	ret = VL53L0X_StaticInit(&drv_data->vl53l0x);
```
Missed this before. You don't check the return value for this function.

VVESTM replied:
You're right ! Fixed

VVESTM commented:
Hi,

I just rebased on the master branch because of the same conflict on ext/hal/Makefile.
No change at all compared to previous version.

@erwango, @bogdan-davidoaia, @dbkinder, @nashif  or @galak, is it possible to merge this pull request ?

Thanks a lot for your help.
Vincent

pfalcon commented:
@VVESTM : The master is currently in the hard release freeze for 1.9. Hopefully will be open for merges again next week.

VVESTM commented:
Hi,

I rebased this morning because of a merge conflict. Now, I have an error in automatic test but this is not linked to my modifications... (error in drivers/sensor/bmi160)

erwango commented on ext/hal/Kconfig:
```diff
@@ -22,6 +22,10 @@ source "ext/hal/qmsi/Kconfig"
 
+source "ext/hal/st/lib/Kconfig"
+
+source "ext/hal/ti/cc3200sdk/Kconfig"
```
Remove this

erwango commented on ext/hal/Makefile:
```diff
@@ -13,4 +13,8 @@ endif
 
+-include $(srctree)/ext/hal/st/lib/Makefile
+
+-include $(srctree)/ext/hal/ti/cc3200sdk/Makefile
```
Remove this

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+	u8_t PhaseCal;
+	u32_t refSpadCount;
+	u8_t isApertureSpads;
+	struct vl53l0x_data *drv_data = dev->driver_data;
```
Can you revert the order of all these lines?

Usually (just for readability), driver data is set the first.

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+	u8_t ret;
+	u16_t vl53l0x_id = 0;
+	VL53L0X_DeviceInfo_t VL53L0X_DeviceInfo;
+	struct device *gpio = device_get_binding(CONFIG_VL53L0X_GPIO_DEV_NAME);
```
actually, declare gpio here, but assign it right before the gpio validity test. it's ugly as it is right now.

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.h:
```diff
@@ -0,0 +1,34 @@
+#define VL53L0X_SETUP_SIGMA_LIMIT          (60*65536)
+#define VL53L0X_SETUP_MAX_TIME_FOR_RANGING     33000
+#define VL53L0X_SETUP_PRE_RANGE_VCSEL_PERIOD   18
+#define VL53L0X_SETUP_FINAL_RANGE_VCSEL_PERIOD 14
```
Can you reference if possible all these numbers above? (datasheet chapter/section will do)

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+	/* Get info from sensor */
+	memset(&VL53L0X_DeviceInfo, 0, sizeof(VL53L0X_DeviceInfo_t));
+
+	ret = VL53L0X_GetDeviceInfo(&(drv_data->vl53l0x), &VL53L0X_DeviceInfo);
```
no need of the () around drv_data->vl5310x

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+	struct vl53l0x_data *drv_data = dev->driver_data;
+	u8_t ret;
+	u16_t vl53l0x_id = 0;
+	VL53L0X_DeviceInfo_t VL53L0X_DeviceInfo;
```
don't use camel case on your own variable. 
vl5310x_dev_info will do.

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+
+static int vl53l0x_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
```
you don't need to cast driver_data explicitly as it is a void*, that's a detail though.

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+		return -EINVAL;
+	}
+
+	return ret;
```
return 0;

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+static int vl53l0x_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
+	u8_t ret = 0;
```
no need of initializing ret here.

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+			|| chan == SENSOR_CHAN_PROX);
+
+	val->val1 = drv_data->RangingMeasurementData.RangeMilliMeter;
+	val->val2 = 0;
```
hum, I guess you need to put these 2 lines in an else after if (chan == SENSOR_CHAN_PROX) {}

It's useless to assign val1 if it's meant to be overwritten in case of chan being SENSOR_CHAN_PROX

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+
+	ret = VL53L0X_StaticInit(&drv_data->vl53l0x);
+	if (ret) {
+		SYS_LOG_DBG("VL53L0X_StaticInit failed");
```
SYS_LOG_ERR()

tbursztyka commented on drivers/sensor/vl53l0x/vl53l0x.c:
```diff
@@ -0,0 +1,254 @@
+					    &VhvSettings,
+					    &PhaseCal);
+	if (ret) {
+		SYS_LOG_DBG("VL53L0X_PerformRefCalibration failed");
```
SYS_LOG_ERR()

tbursztyka reviewed (changes requested):
Isn't the order of commits a bit weird? it seems shuffled to me. I would see first:
1 - sensor.h patch
2 - ext/hal patch
3 - driver patch
4 - board patch
5 - sample patch

Anyway, on vl5310x_init(), you'll need to revise the error codes returned. -EINVAL is valid for invalid parameter (like if gpio is null, it is the right error code). But fox chip id verification for instance I would expect something more relevant, like -ENODEV maybe?

Please check.

VVESTM commented:
New version including remarks from @erwango and @tbursztyka

galak reviewed (changes requested):
for the import of the lib code, can you separate the original code from the build system changes

VVESTM commented:
Update following @galak remark.
I splitted the ext/hal/st commit in two parts : pure ST lib part and update of the build system.

galak reviewed (changes requested):
So the 'ext: Add official ST library for vl53l0x' should not have any Kconfig, Kbuild, or Makefile modifications, its should pretty much be the import of the source in a pristine way and update to a README.  Also the commit message should have details as outlined in https://github.com/zephyrproject-rtos/zephyr/blob/master/CONTRIBUTING.rst

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/core/src/vl53l0x_api.c:
```diff
@@ -382,10 +382,6 @@ VL53L0X_Error VL53L0X_DataInit(VL53L0X_DEV Dev)
-	/* Set I2C standard mode */
-	if (Status == VL53L0X_ERROR_NONE)
-		Status = VL53L0X_WrByte(Dev, 0x88, 0x00);
-
```
this seems concerning, wondering why it had to be removed.

VVESTM replied:
Good hint. I had issue with this part while performing my developments. Now, the sensor is working OK with those lines. So I will reintroduce them in my next version.

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/inc/vl53l0x_platform.h:
```diff
@@ -62,6 +62,7 @@ typedef struct {
     uint8_t   I2cDevAddr;                /*!< i2c device address user specific field */
     uint8_t   comms_type;                /*!< Type of comms : VL53L0X_COMMS_I2C or VL53L0X_COMMS_SPI */
     uint16_t  comms_speed_khz;           /*!< Comms speed [kHz] : typically 400kHz for I2C           */
+    struct device *i2c;
```
this doesn't seem like it should be required or how we support things.

VVESTM replied:
Not sure I well understood your remark here. This i2c pointer is used here to store the pointer on i2c device. This is obtained by a call to device_get_binding(CONFIG_VL53L0X_I2C_MASTER_DEV_NAME) in the driver and then, I use it on the lib part to deal with i2c.
In my opinion, we need it. Perhaps you prefer to handle this in an other way ?

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c:
```diff
@@ -101,44 +109,43 @@ VL53L0X_Error VL53L0X_UnlockSequenceAccess(VL53L0X_DEV Dev){
+         SYS_LOG_ERR("Failed to write");
+    }
 
     return Status;
```
why modify all this, why not just provide a version of VL53L0X_write_multi?

VVESTM replied:
The adaptation of the lib to Zephyr has to be done in ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c. That's why I call Zephyr function from here.
The usage of VL53L0X_write_multi is not a good option for me. It is part of vl53l0x_i2c_win_serial_comms.c file, which is present in ST delivery but this is just presented as an example. When reading the doc, they explain that we need to do the adaptation in vl53l0x_platform.c.

Regarding this, are you OK to keep my implementation ?

galak replied:
if you feel like you need a full implementation of vl53l0x_platform.c than do it in the zephyr driver, not here.

VVESTM replied:
Done in the last version. I have moved this file to driver and cleaned it.

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c:
```diff
@@ -101,44 +109,43 @@ VL53L0X_Error VL53L0X_UnlockSequenceAccess(VL53L0X_DEV Dev){
+        SYS_LOG_ERR("Failed to read");
+        return -EIO;
+    }
 
```
why modify all this, why not just provide a version of VL53L0X_read_multi?

VVESTM replied:
same as previous remark

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c:
```diff
@@ -147,44 +154,51 @@ VL53L0X_Error VL53L0X_ReadMulti(VL53L0X_DEV Dev, uint8_t index, uint8_t *pdata,
+        Status = VL53L0X_ERROR_CONTROL_INTERFACE;
+        SYS_LOG_ERR("i2c_reg_write_byte failed (%d)", Status);
+    }
 
```
why modify all this, why not just provide a version of VL53L0X_write_byte?

VVESTM replied:
same as previous remark

galak reviewed (changes requested):
So it seems there are a number of functions you modify in vl53l0x_platform.c, instead we should provide implementations of the lower level functions that code is calling that does the zephyr specifics.

VVESTM commented:
Hello @galak, I have pushed a new version including some of your remarks. For other remarks, please see my inline answer in your comment.
Regards

VVESTM commented:
I updated a new version following @galak review. Now, the vl53l0x_platform.c file is stored in the driver directory.

VVESTM committed:
ext: enable st/lib compilation

This commit modify the ext/ build system to add
the ST Microelectronics libraries.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM commented:
New version including cmake.

SebastianBoe commented on drivers/sensor/vl53l0x/CMakeLists.txt:
```diff
@@ -0,0 +1,2 @@
+zephyr_sources_ifdef(CONFIG_VL53L0X vl53l0x.c)
+zephyr_sources_ifdef(CONFIG_VL53L0X vl53l0x_platform.c)
```
You should omit the ifdef because this entire directory is already ifdef'ed on CONFIG_VL53L0X.

Ifdef'ing on VL53L0X in here might communicate to the reader that VL53L0X is not always defined when processing this CMakeLists.txt.

So

zephyr_sources(
  vl53l0x.c
  vl53l0x_platform.c
)

VVESTM replied:
Thanks Sebastian, good remark !
I have integrated your modification.

galak replied:
Are you going to change the CMakeLists.txt?

SebastianBoe reviewed (commented):
Trying to change my +1 into a 0 ...

galak commented on drivers/sensor/vl53l0x/vl53l0x_platform.c:
```diff
@@ -0,0 +1,216 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+********************************************************************************/
+
```
Can the platform driver be apache licensed?

galak commented on ext/hal/st/lib/sensor/vl53l0x/api/platform/src/vl53l0x_platform.c:
```diff
@@ -1,275 +0,0 @@
-/*******************************************************************************
```
lets not remove the example code

VVESTM replied:
Personally, I prefer to remove it in order not to have 2 files that can be confusing for the user. If user want the original file, it is in git history or on ST website.

galak commented on drivers/sensor/vl53l0x/vl53l0x_platform.c:
```diff
@@ -0,0 +1,216 @@
+	return status;
+}
+
+int abs(int i)
```
can we move abs() into ext/hal/st/lib/sensor/vl53l0x somewhere?

VVESTM replied:
fixed by using newlibc...

galak reviewed (changes requested):
Some things to deal with, but looking better.

galak reviewed (changes requested):
Looks like the drivers/sensor/vl53l0x/vl53l0x_platform.c is a non-unix text file

galak commented on ext/hal/st/lib/README:
```diff
@@ -0,0 +1,8 @@
+
+Directory contains all the official libraries provided by STMicroelectronics.
+
+Available libs : 
```
remove space after :

galak reviewed (commented):
Can we not use sensor/vl53l0x/api/platform or sensor/vl53l0x/api/platform/inc... such that we get all platform code & header from just drivers/sensor/vl53l0x

VVESTM committed:
ext: Add official ST library for vl53l0x

In order to ease the usage of its sensors, STMicroelectronics provides
some generic libraries. The first sensor to use that is vl53l0x time of
flight sensor. I have made the implementation generic enough to add some
more libraries in the future.

All the libraries will be located in ext/hal/st/lib.

The vl53l0X library is made of 2 parts :
    * the core, that is generic to all platforms
    * the platform, contains the adaptation layer. This will be
      implemented in driver as it is Zephyr specific.

Origin: ST Microelectronics
License: BSD-3-Clause
URL: http://www.st.com/en/embedded-software/stsw-img005.html
Commit: 1.0.2
Purpose: API to ease the usage of vl53l0x sensor
Maintained-by: External

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM committed:
ext: Fix warnings in vl53l0x ST library when compiling on Zephyr

Warning are present in the default ST source code, fix them here.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM committed:
drivers: sensor: vl53l0x: ST library customization for Zephyr

The generic library provided by ST in ext/hal/st/lib/sensor/vl53l0x
needs to be adapted to the platform. In this case, the main
modification is the way to use I2C. This is now using the Zephyr
I2C API.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM commented:
New version including latest comments, except license issue...

VVESTM commented:
I have fixed the license issue for all files in driver. It is now Apache 2.0 as requested.
Should be OK now

galak commented:
@erwango Would like this to only support I2C dts once that goes in. Can you help @VVESTM with the changes needed for that.  Thanks

galak commented:
I've merged the I2C device tree support, so please update this against master and make the changes to support dts.

VVESTM committed:
sample: sensor: Add vl53l0x sample application

This sample application allows to test vl53l0x distance sensor.
By default it is built with disco_l475_iot1 board as it is
built-in.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

VVESTM commented:
Hi,
New version including device tree support. And all tests are passed

erwango commented:
@galak : Do you think we'd be ready for merge on this one once this is rebased?

galak commented on drivers/sensor/vl53l0x/vl53l0x_i2c_platform.h:
```diff
@@ -0,0 +1,378 @@
+#include <stdint.h>
+#include <stdarg.h>
+
+#define	   I2C                0x01
```
How much of anything in this file is used?  It seems as if none of it is - if so can we remove it?

galak commented on drivers/sensor/vl53l0x/vl53l0x_platform.c:
```diff
@@ -0,0 +1,210 @@
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SENSOR_LEVEL
+#include <logging/sys_log.h>
+
+VL53L0X_Error VL53L0X_LockSequenceAccess(VL53L0X_DEV Dev)
```
Can we remove VL53L0X_LockSequenceAccess, don't see it used anywhere

galak commented on drivers/sensor/vl53l0x/vl53l0x_platform.c:
```diff
@@ -0,0 +1,210 @@
+	return Status;
+}
+
+VL53L0X_Error VL53L0X_UnlockSequenceAccess(VL53L0X_DEV Dev)
```
Can we remove VL53L0X_UnlockSequenceAccess, don't see it used anywhere.

galak commented on drivers/sensor/vl53l0x/vl53l0x.h:
```diff
@@ -0,0 +1,43 @@
+ * There are also examples of use in the L4 cube FW
+ *   (http://www.st.com/en/embedded-software/stm32cubel4.html)
+ */
+#define VL53L0X_I2C_ADDR       0x29
```
ideally we'd get VL53L0X_I2C_ADDR from device tree defines.

galak commented on samples/sensor/vl53l0x/prj.conf:
```diff
@@ -0,0 +1,12 @@
+CONFIG_VL53L0X=y
+CONFIG_VL53L0X_PROXIMITY_THRESHOLD=100
+
+CONFIG_NEWLIB_LIBC=y
```
Is NEWLIB_LIBC really needed?

VVESTM replied:
This one is needed for abs function

pfalcon replied:
abs can be a one-liner macro.

VVESTM replied:
yes, I used it before in the first versions of this PR...
For an other topic, I discovered NEWLIB_LIBC and see that it include abs. So my conclusion was to use the official way instead of redefining existing functions.

If this is the last blocking point, I can reimplement this function. Else, I prefer to keep NEWLIB_LIBC.

pfalcon replied:
Nope, just a thought, not blocking, can be left for later, if someone ever has a real issue with it.

galak commented on drivers/sensor/vl53l0x/vl53l0x.h:
```diff
@@ -0,0 +1,43 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SENSOR_VL53L0X_H__
```
As this header is only used by vl53l0x.c, can we just put the defines & struct into vl53l0x.c and remove this file.

galak reviewed (changes requested):
A few changes / cleanups.

galak commented on drivers/sensor/vl53l0x/Kconfig:
```diff
@@ -0,0 +1,59 @@
+	  Specify the device name of the I2C master device to which VL53L0X is
+	  connected.
+
+config VL53L0X_GPIO_DEV_NAME
```
Can we rename this to VL53L0X_XSHUT_GPIO_DEV_NAME

galak commented on drivers/sensor/vl53l0x/Kconfig:
```diff
@@ -0,0 +1,59 @@
+	  The device name of the GPIO device to which the VL53L0X xshut pin
+	  is connected.
+
+config VL53L0X_GPIO_PIN_NUM
```
Can we rename this to VL53L0X_XSHUT_GPIO_PIN_NUM

galak reviewed (changes requested):
some minor names changes for GPIO kconfig symbols.

VVESTM commented:
New version following @galak comments

galak commented on drivers/sensor/vl53l0x/vl53l0x_platform_log.h:
```diff
@@ -0,0 +1,60 @@
+	TRACE_MODULE_ALL               = 0x7fffffff /* all bits except sign */
+};
+
+#define VL53L0X_ErrLog(...) (void)0
```
lets remove VL53L0X_ErrLog - don't see it referenced anywhere.

galak commented on samples/sensor/vl53l0x/prj.conf:
```diff
@@ -0,0 +1,12 @@
+CONFIG_VL53L0X=y
+CONFIG_VL53L0X_PROXIMITY_THRESHOLD=100
+
+CONFIG_NEWLIB_LIBC=y
```
I agree with paul, lets add a patch in here, to introduce abs into lib/libc/minimal/include/stdlib.h

something like:

#define abs(x) ((x)<0 ? -(x) : (x))

This removes the NEWLIB dependency.

galak reviewed (changes requested):
1. Lets introduce abs() into lib/libc/minimal/include/stdlib.h to remove NEWLIB dev
2. Are we going to fix drivers/sensor/vl53l0x/CMakeLists.txt?
3. Remove VL53L0X_ErrLog as not used anywhere

VVESTM committed:
lib: add abs function to minimal libc

Add abs function to the minimal libc. This is present in
NEWLIB_LIBC, but adding it here avoid to make a dependency
with NEWLIB_LIBC.

Signed-off-by: Vincent Veron <vincent.veron@st.com>

pfalcon commented:
Sorry for missing it for few days, but there's a proposal to switch SENSOR_CHAN_DISTANCE units to meters instead of millimeters:
https://github.com/zephyrproject-rtos/zephyr/issues/5693#issuecomment-361430731

Comments welcome.
